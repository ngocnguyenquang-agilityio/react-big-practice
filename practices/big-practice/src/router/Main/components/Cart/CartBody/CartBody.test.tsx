import { RenderResult, render, screen } from '@testing-library/react';

// Components
import CartBody from '.';
import { useCartStore } from '@stores/cartStore';

jest.mock('@stores/cartStore');
const useMockCartStore = jest.mocked(useCartStore);

describe('CartBody Component', () => {
  let comp: RenderResult;
  beforeEach(() => {
    useMockCartStore.mockReturnValue({
      cartItems: [
        {
          id: 1,
          title: 'Product Detail',
          price: 10,
          thumbnail:
            'https://demo.vercel.store/_next/image?url=https%3A%2F%2Fcdn.shopify.com%2Fs%2Ffiles%2F1%2F0754%2F3727%2F7491%2Ffiles%2Ft-shirt-1.png%3Fv%3D1689798965&w=3840&q=75',
          category: 'Product',
          quantity: 1,
        },
        {
          id: 2,
          title: 'Product Detail 2',
          price: 20,
          thumbnail:
            'https://demo.vercel.store/_next/image?url=https%3A%2F%2Fcdn.shopify.com%2Fs%2Ffiles%2F1%2F0754%2F3727%2F7491%2Ffiles%2Ft-shirt-1.png%3Fv%3D1689798965&w=3840&q=75',
          category: 'Product',
          quantity: 3,
        },
      ],
    });
    comp = render(<CartBody />);
  });
  test('Should render correctly', () => {
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    expect(comp).toMatchSnapshot();
  });

  test('Should render with no cartItems', () => {
    useMockCartStore.mockReturnValue({
      cartItems: [],
    });
    comp = render(<CartBody />);
    expect(screen.getByText('Your cart is empty.')).toBeTruthy();
  });
});
