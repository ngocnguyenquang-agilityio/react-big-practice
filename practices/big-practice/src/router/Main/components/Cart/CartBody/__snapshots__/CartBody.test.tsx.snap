// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartBody Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-full flex-col justify-between overflow-hidden p-1"
      >
        <ul
          class="flex-grow overflow-auto py-4"
        >
          <li
            class="flex w-full flex-col border-b border-neutral-700"
          >
            <div
              class="relative flex w-full flex-row justify-between px-1 py-4"
            >
              <div
                class="absolute z-40 -mt-2 ml-[55px]"
              >
                <button
                  class="ease flex h-[17px] w-[17px] items-center justify-center rounded-full bg-neutral-500 transition-all duration-200"
                >
                  <img
                    alt="remove-icon"
                    height="16"
                    src=""
                    width="16"
                  />
                </button>
              </div>
              <a
                class="z-30 flex flex-row space-x-4"
                href=""
              >
                <div
                  class="relative h-16 w-16 cursor-pointer overflow-hidden rounded-md border border-neutral-700 bg-neutral-900 hover:bg-neutral-800"
                >
                  <img
                    class="h-full w-full object-cover"
                    loading="lazy"
                    src="https://demo.vercel.store/_next/image?url=https%3A%2F%2Fcdn.shopify.com%2Fs%2Ffiles%2F1%2F0754%2F3727%2F7491%2Ffiles%2Ft-shirt-1.png%3Fv%3D1689798965&w=3840&q=75"
                  />
                </div>
                <div
                  class="flex flex-1 flex-col text-base"
                >
                  <span
                    class="leading-tight"
                  >
                    Product Detail
                  </span>
                  <p
                    class="text-sm text-neutral-400"
                  >
                    Black / 7 x 9 inch
                  </p>
                </div>
              </a>
              <div
                class="flex h-16 flex-col justify-between"
              >
                <p
                  class="flex justify-end space-y-2 text-right text-sm"
                >
                  10
                  <span
                    class="ml-1 inline"
                  >
                    USD
                  </span>
                </p>
                <div
                  class="ml-auto flex h-9 flex-row items-center rounded-full border border-neutral-700"
                >
                  <button
                    class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80 ml-auto cursor-not-allowed"
                    disabled=""
                  >
                    <img
                      alt="minus-icon"
                      height="16"
                      src=""
                      width="16"
                    />
                  </button>
                  <p
                    class="w-6 text-center"
                  >
                    <span
                      class="w-full text-sm text-white"
                    >
                      1
                    </span>
                  </p>
                  <button
                    class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80"
                  >
                    <img
                      alt="plus-icon"
                      height="16"
                      src=""
                      width="16"
                    />
                  </button>
                </div>
              </div>
            </div>
          </li>
          <li
            class="flex w-full flex-col border-b border-neutral-700"
          >
            <div
              class="relative flex w-full flex-row justify-between px-1 py-4"
            >
              <div
                class="absolute z-40 -mt-2 ml-[55px]"
              >
                <button
                  class="ease flex h-[17px] w-[17px] items-center justify-center rounded-full bg-neutral-500 transition-all duration-200"
                >
                  <img
                    alt="remove-icon"
                    height="16"
                    src=""
                    width="16"
                  />
                </button>
              </div>
              <a
                class="z-30 flex flex-row space-x-4"
                href=""
              >
                <div
                  class="relative h-16 w-16 cursor-pointer overflow-hidden rounded-md border border-neutral-700 bg-neutral-900 hover:bg-neutral-800"
                >
                  <img
                    class="h-full w-full object-cover"
                    loading="lazy"
                    src="https://demo.vercel.store/_next/image?url=https%3A%2F%2Fcdn.shopify.com%2Fs%2Ffiles%2F1%2F0754%2F3727%2F7491%2Ffiles%2Ft-shirt-1.png%3Fv%3D1689798965&w=3840&q=75"
                  />
                </div>
                <div
                  class="flex flex-1 flex-col text-base"
                >
                  <span
                    class="leading-tight"
                  >
                    Product Detail 2
                  </span>
                  <p
                    class="text-sm text-neutral-400"
                  >
                    Black / 7 x 9 inch
                  </p>
                </div>
              </a>
              <div
                class="flex h-16 flex-col justify-between"
              >
                <p
                  class="flex justify-end space-y-2 text-right text-sm"
                >
                  20
                  <span
                    class="ml-1 inline"
                  >
                    USD
                  </span>
                </p>
                <div
                  class="ml-auto flex h-9 flex-row items-center rounded-full border border-neutral-700"
                >
                  <button
                    class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80 ml-auto false"
                  >
                    <img
                      alt="minus-icon"
                      height="16"
                      src=""
                      width="16"
                    />
                  </button>
                  <p
                    class="w-6 text-center"
                  >
                    <span
                      class="w-full text-sm text-white"
                    >
                      3
                    </span>
                  </p>
                  <button
                    class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80"
                  >
                    <img
                      alt="plus-icon"
                      height="16"
                      src=""
                      width="16"
                    />
                  </button>
                </div>
              </div>
            </div>
          </li>
        </ul>
        <div
          class="py-4 text-sm text-neutral-400"
        >
          <div
            class="mb-3 flex items-center justify-between border-b pb-1 border-neutral-700"
          >
            <p>
              Taxes
            </p>
            <p
              class="text-right text-base text-white"
            >
              $0.00
              <span
                class="ml-1 inline"
              >
                USD
              </span>
            </p>
          </div>
          <div
            class="mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700"
          >
            <p>
              Shipping
            </p>
            <p
              class="text-right"
            >
              Calculated at checkout
            </p>
          </div>
          <div
            class="mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700"
          >
            <p>
              Total
            </p>
            <p
              class="text-right text-base text-white"
            >
              $
              70
              <span
                class="ml-1 inline"
              >
                USD
              </span>
            </p>
          </div>
          <button
            class="inline-flex items-center text-white justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 bg-blue-600 hover:opacity-90 disabled:cursor-not-allowed disabled:opacity-50 w-full h-11 rounded-full"
          >
            Proceed to Checkout
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-full flex-col justify-between overflow-hidden p-1"
    >
      <ul
        class="flex-grow overflow-auto py-4"
      >
        <li
          class="flex w-full flex-col border-b border-neutral-700"
        >
          <div
            class="relative flex w-full flex-row justify-between px-1 py-4"
          >
            <div
              class="absolute z-40 -mt-2 ml-[55px]"
            >
              <button
                class="ease flex h-[17px] w-[17px] items-center justify-center rounded-full bg-neutral-500 transition-all duration-200"
              >
                <img
                  alt="remove-icon"
                  height="16"
                  src=""
                  width="16"
                />
              </button>
            </div>
            <a
              class="z-30 flex flex-row space-x-4"
              href=""
            >
              <div
                class="relative h-16 w-16 cursor-pointer overflow-hidden rounded-md border border-neutral-700 bg-neutral-900 hover:bg-neutral-800"
              >
                <img
                  class="h-full w-full object-cover"
                  loading="lazy"
                  src="https://demo.vercel.store/_next/image?url=https%3A%2F%2Fcdn.shopify.com%2Fs%2Ffiles%2F1%2F0754%2F3727%2F7491%2Ffiles%2Ft-shirt-1.png%3Fv%3D1689798965&w=3840&q=75"
                />
              </div>
              <div
                class="flex flex-1 flex-col text-base"
              >
                <span
                  class="leading-tight"
                >
                  Product Detail
                </span>
                <p
                  class="text-sm text-neutral-400"
                >
                  Black / 7 x 9 inch
                </p>
              </div>
            </a>
            <div
              class="flex h-16 flex-col justify-between"
            >
              <p
                class="flex justify-end space-y-2 text-right text-sm"
              >
                10
                <span
                  class="ml-1 inline"
                >
                  USD
                </span>
              </p>
              <div
                class="ml-auto flex h-9 flex-row items-center rounded-full border border-neutral-700"
              >
                <button
                  class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80 ml-auto cursor-not-allowed"
                  disabled=""
                >
                  <img
                    alt="minus-icon"
                    height="16"
                    src=""
                    width="16"
                  />
                </button>
                <p
                  class="w-6 text-center"
                >
                  <span
                    class="w-full text-sm text-white"
                  >
                    1
                  </span>
                </p>
                <button
                  class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80"
                >
                  <img
                    alt="plus-icon"
                    height="16"
                    src=""
                    width="16"
                  />
                </button>
              </div>
            </div>
          </div>
        </li>
        <li
          class="flex w-full flex-col border-b border-neutral-700"
        >
          <div
            class="relative flex w-full flex-row justify-between px-1 py-4"
          >
            <div
              class="absolute z-40 -mt-2 ml-[55px]"
            >
              <button
                class="ease flex h-[17px] w-[17px] items-center justify-center rounded-full bg-neutral-500 transition-all duration-200"
              >
                <img
                  alt="remove-icon"
                  height="16"
                  src=""
                  width="16"
                />
              </button>
            </div>
            <a
              class="z-30 flex flex-row space-x-4"
              href=""
            >
              <div
                class="relative h-16 w-16 cursor-pointer overflow-hidden rounded-md border border-neutral-700 bg-neutral-900 hover:bg-neutral-800"
              >
                <img
                  class="h-full w-full object-cover"
                  loading="lazy"
                  src="https://demo.vercel.store/_next/image?url=https%3A%2F%2Fcdn.shopify.com%2Fs%2Ffiles%2F1%2F0754%2F3727%2F7491%2Ffiles%2Ft-shirt-1.png%3Fv%3D1689798965&w=3840&q=75"
                />
              </div>
              <div
                class="flex flex-1 flex-col text-base"
              >
                <span
                  class="leading-tight"
                >
                  Product Detail 2
                </span>
                <p
                  class="text-sm text-neutral-400"
                >
                  Black / 7 x 9 inch
                </p>
              </div>
            </a>
            <div
              class="flex h-16 flex-col justify-between"
            >
              <p
                class="flex justify-end space-y-2 text-right text-sm"
              >
                20
                <span
                  class="ml-1 inline"
                >
                  USD
                </span>
              </p>
              <div
                class="ml-auto flex h-9 flex-row items-center rounded-full border border-neutral-700"
              >
                <button
                  class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80 ml-auto false"
                >
                  <img
                    alt="minus-icon"
                    height="16"
                    src=""
                    width="16"
                  />
                </button>
                <p
                  class="w-6 text-center"
                >
                  <span
                    class="w-full text-sm text-white"
                  >
                    3
                  </span>
                </p>
                <button
                  class="text-white text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 py-2 ease flex h-full min-w-[36px] max-w-[36px] flex-none items-center justify-center rounded-full px-2 transition-all duration-200 hover:border-neutral-800 hover:opacity-80"
                >
                  <img
                    alt="plus-icon"
                    height="16"
                    src=""
                    width="16"
                  />
                </button>
              </div>
            </div>
          </div>
        </li>
      </ul>
      <div
        class="py-4 text-sm text-neutral-400"
      >
        <div
          class="mb-3 flex items-center justify-between border-b pb-1 border-neutral-700"
        >
          <p>
            Taxes
          </p>
          <p
            class="text-right text-base text-white"
          >
            $0.00
            <span
              class="ml-1 inline"
            >
              USD
            </span>
          </p>
        </div>
        <div
          class="mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700"
        >
          <p>
            Shipping
          </p>
          <p
            class="text-right"
          >
            Calculated at checkout
          </p>
        </div>
        <div
          class="mb-3 flex items-center justify-between border-b pb-1 pt-1 border-neutral-700"
        >
          <p>
            Total
          </p>
          <p
            class="text-right text-base text-white"
          >
            $
            70
            <span
              class="ml-1 inline"
            >
              USD
            </span>
          </p>
        </div>
        <button
          class="inline-flex items-center text-white justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 bg-blue-600 hover:opacity-90 disabled:cursor-not-allowed disabled:opacity-50 w-full h-11 rounded-full"
        >
          Proceed to Checkout
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
